sockeye/beam_search.py

```diff
@@ -388,25 +396,32 @@ def hybrid_forward(self, F, scores, target_dists, finished, best_hyp_indices):
         return best_hyp_indices, best_word_indices, values
 
 
-def _repeat_states(states: List, beam_size) -> List:
+def _repeat_states(states: List, beam_size, state_structure) -> List:
     repeated_states = []
-    for i in range(len(states)):
-        if i < 2:
+    for i in range(len(state_structure)):
+        if state_structure[i] == 's':
+            # Steps and source_bias have batch dimension on axis 0
             state = states[i].repeat(repeats=beam_size, axis=0)
         else:
```

Comment:
better safe than sorry and check for `format == 'e' or format == 'd'`, and raise a ValueError otherwise.

I'd also suggest setting a variable `axis` based on the if statement and then have only a single 'repeat' line.